website/docs/r/nat_gateway.html.markdown

```diff
@@ -0,0 +1,88 @@
+---
+subcategory: ""
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_nat_gateway"
+sidebar_current: "docs-azurerm-resource-nat-gateway"
+description: |-
+  Manages an Azure NAT Gateway instance.
+---
+# azurerm_nat_gateway
+
+Manages an Azure NAT Gateway instance.
+
+-> **NOTE:** The Azure NAT Gateway service is currently in private preview. Your subscription must be on the NAT Gateway private preview whitelist for this resource to be provisioned correctly. If you attempt to provision this resource and receive an `InvalidResourceType` error that means that your subscription is not part of the NAT Gateway private preview whitelist and you will not be able to use this resource. The NAT Gateway private preview service is currently only available in the `East US 2` and `West Central US` regions.

```

Comment:
Could we include warnings about possible breaking changes & a note to contact azure support if a user wants to participate?

Reply:
Added some more information, please have a look